app/src/main/java/by/paranoidandroid/serviceexample/services/HardJobService.java

```diff
@@ -0,0 +1,86 @@
+package by.paranoidandroid.serviceexample.services;
+
+import android.app.Service;
+import android.content.Intent;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.IBinder;
+import android.os.Looper;
+import android.os.Message;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+
+public class HardJobService extends Service {
```

Comment:
If your app targets API level 26 or higher, the system imposes restrictions on running background services when the app itself isn't in the foreground. In most cases like this, your app should use a scheduled job instead.